screeps-game-api/src/objects/mod.rs

```diff
@@ -190,16 +190,22 @@ impl HasPosition for RoomPosition {
     }
 }
 
-/// All RoomObjects have positions.
-impl<T> HasPosition for T
-where
-    T: RoomObjectProperties,
-{
-    fn pos(&self) -> RoomPosition {
-        js_unwrap!(@{self.as_ref()}.pos)
+/// Trait covering all objects with an id.
+pub trait HasId : AsRef<Reference> {
```

Comment:
I think it's safe to have this trait inherit from `RoomObjectProperties` too, since to my knowledge there is nothing with an ID which is not also a `RoomObject`.

Reply:
You are correct!